### DISCUSSION TITLE
Pull primary key from database on insert if generated elsewhere

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#5267] [Feature Requests]
In certain database vendors, you can use functions to generate values for primary keys (like `gen_random_uuid()` under Postgres). However, Directus isn't aware of these function-based values, and can't reliably extract the generated value after the insert.

Some databases support `RETURNING` on insert, but this doesn't seem to be available across the board.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
